### ISSUE TITLE
multiple webview case 

### ISSUE THREAD

**Issue by nick0504k** [gronxb/webview-bridge#43]
Hi,

Thank you for this great project.  I’m struggling with handling multiple web views using linkWebMethod and shared state. I know there’s InAppBrowser, but it doesn’t offer the full customization I need.

Do you have any tips or advice for handling multiple webviews?

**Comment by gronxb** [comment]
hello !

can you share a small codebase ?
please share a detailed scenario.

If you need a repro, go to https://github.com/gronxb/webview-bridge-example/tree/main
You can prepare it here.

**Comment by nick0504k** [comment]
Here’s a sample codebase. Please let me know if you need any additional information or if there’s anything unclear.


### React Native

App.tsx
```
<NavigationContainer ref={RootNavigation.navigationRef}>
      <Stack.Navigator>
        <Stack.Group screenOptions={defaultScreenOptions}>
          <Stack.Screen name={ROUTES.WEB_VIEW} component={WebViewScreen} />
      </Stack.Navigator>
    </NavigationContainer>
```


WebViewScreen.tsx
```
export const WebViewScreen: React.FC<Props> = (props) => {
const { params } = props.route;
 const webView = useRef<BridgeWebView>(null);

const { data, setDataText } = useBridge(appBridge);
 return (
    <WebView
      originWhitelist={['*']}
      source={ source={{ uri: params?.uri }}
      javaScriptEnabled
      domStorageEnabled
      ref={webView}
      onMessage={onWebViewMessage}
      onNavigationStateChange={onNavigationStateChange}
      webviewDebuggingEnabled
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      injectedJavaScript={webviewLocalStorage}
      mixedContentMode="compatibility"
      geolocationEnabled
      {...rest}
    />
    <Text
            style={{
              marginBottom: 10,
              textAlign: "center",
            }}
          >
        Native Data Text: {data.text}
      </Text>
      <TextInput
        value={data.text}
        onChangeText={setDataText}
        style={{ borderWidth: 1, minWidth: "50%", maxWidth: "50%" }}
      />
  );
```

bridge.ts in react native 

```

export const appBridge = bridge<AppBridgeState>(({ get, set }) => ({
  async goBack() {
    if (navigationRef.current?.isReady()) {
      navigationRef.current.goBack();
    }
  },
 async push<RouteName extends keyof RootStackParamList>(
    name: RouteName,
    params: RootStackParamList[RouteName],
  ) {
    if (navigationRef.current?.isReady()) {
      navigationRef.current.dispatch(StackActions.push(name, params));
    }
  },
  data: {
    text: "",
  },
  async setDataText(text) {
      set({
        data: {
          text,
        },
      });
    },
  }));

export type AppBridge = typeof appBridge;

export const { WebView, linkWebMethod } = createWebView({
  bridge: appBridge,
  debug: false,
  fallback: (method) => {
    console.warn(`Method '${method}' not found in native`);
  },
});
```


###  React 
 ```
 const text = useBridge(bridge.store, (state) => state.data.text);
 
  
      return (
        <div>
          <button
            onClick={() => { bridge.push('WebViewScreen', {uri: 'localhost:3000/home'} )}>
          <p>Native Data Text: {text}</p>
          <input
            type="text"
            value={text}
            onChange={(e) => bridge.setDataText(e.target.value)}
          />
        </div>
     )


```


Initially, changing the state in React and React Native works well. However, when I push a new ‘WebViewScreen’ and then pop it, state sharing or the linkWebMethod doesn’t function as expected.

**Comment by gronxb** [comment]
Thank you for your report.

This has been resolved in 1.4.4.

as-is:

https://github.com/gronxb/webview-bridge/assets/41789633/c32a038a-9bf3-42c7-b290-b46b4006761f

to-be:

https://github.com/gronxb/webview-bridge/assets/41789633/ddb8345c-c417-4d9b-b7f7-643f37ea9089

If there are any unresolved issues, please feel free to reopen the case at any time.

**Comment by nick0504k** [comment]
Thank you so much for your prompt fix! 😊 It really saved me.
